OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -1141,7 +1141,7 @@ void ProductionUnits(Actor self)
 				return;
 
 			// No construction yards - Build a new MCV
-			if (!HasAdequateFact() && !self.World.Actors.Any(a => a.Owner == Player &&
+			if (Info.UnitsCommonNames.Mcv.Count != 0 && !HasAdequateFact() && !self.World.Actors.Any(a => a.Owner == Player &&
```

Comment:
What is the crash?  The `Info.UnitsCommonNames.Mcv.Contains(a.Info.Name)` check should already be stopping this if the list is empty.  If the crash is in `HasAdequateFact` then the fix should be applied in there.

Reply:
It is still reasonable to apply a shortcut check here so that we don't have to walk over all the actors.  Please change it to `!Info.UnitsCommonNames.Mcv.Any()` to fit our style conventions after addressing my question above.

Reply:
Crash was with GetInfoByCommonName below. I'll post the crash log when i get to the PC.

Reply:
Crash was with GetInfoByCommonName below. I'll post the crash log when i get to the PC.